src/main/java/com/iexec/commons/poco/chain/Web3jAbstractService.java

```diff
@@ -261,35 +274,6 @@ public BigInteger getUserGasPrice(float gasPriceMultiplier, long gasPriceCap) {
         return BigInteger.valueOf(Math.min(wishedGasPrice, gasPriceCap));
     }
 
-    /*
-     * This is just a dummy stub for contract reader:
-     * gas price & gas limit is not required when querying (read) an eth node
-     *
-     */
-    public ContractGasProvider getReadingContractGasProvider() {
-        return new ContractGasProvider() {
-            @Override
-            public BigInteger getGasPrice(String contractFunc) {
-                return null;
-            }
-
-            @Override
-            public BigInteger getGasPrice() {
-                return null;
-            }
-
-            @Override
-            public BigInteger getGasLimit(String contractFunc) {
-                return null;
-            }
-
-            @Override
-            public BigInteger getGasLimit() {
-                return null;
-            }
-        };
-    }
-
     public ContractGasProvider getWritingContractGasProvider() {
```

Comment:
Do we still need this public method? I guess `contractGasProvider` should now be retrieved through `getContractGasProvider`.

Reply:
Please, see 228c080